src/sage/graphs/matching_covered_graph.py

```diff
@@ -0,0 +1,1221 @@
+
+    - ``algorithm`` -- string (default: ``'Edmonds'``); the algorithm to be
+      used to compute a maximum matching of the graph among
+
+      - ``'Edmonds'`` selects Edmonds' algorithm as implemented in NetworkX,
+
+      - ``'LP'`` uses a Linear Program formulation of the matching problem.
+
+    - ``solver`` -- string (default: ``None``); specify a Mixed Integer
+      Linear Programming (MILP) solver to be used. If set to ``None``, the
+      default one is used. For more information on MILP solvers and which
+      default solver is used, see the method :meth:`solve
+      <sage.numerical.mip.MixedIntegerLinearProgram.solve>` of the class
+      :class:`MixedIntegerLinearProgram
+      <sage.numerical.mip.MixedIntegerLinearProgram>`.
+
+    - ``verbose`` -- integer (default: ``0``); sets the level of verbosity:
+      set to 0 by default, which means quiet (only useful when ``algorithm
+      == 'LP'``).
+
+    - ``integrality_tolerance`` -- float; parameter for use with MILP
+      solvers over an inexact base ring; see
+      :meth:`MixedIntegerLinearProgram.get_values`.
+
+    OUTPUT:
+
+    - An object of the class :class:`~MatchingCoveredGraph` if the input is
+      valid and the graph is matching covered, or otherwise an error is thrown.
+
+    .. NOTE::
+
+        All remaining arguments are passed to the ``Graph`` constructor
+
+    EXAMPLES:
+
+    Generating an object of the class ``MatchingCoveredGraph`` from the
+    provided instance of ``Graph`` without providing any other information::
+
+        sage: G = MatchingCoveredGraph(graphs.PetersenGraph())
+        sage: G
+        Matching covered petersen graph: graph on 10 vertices
+        sage: G.get_matching()
+        [(0, 5, None), (1, 6, None), (2, 7, None), (3, 8, None), (4, 9, None)]
+
+        sage: G = graphs.StaircaseGraph(4)
+        sage: H = MatchingCoveredGraph(G)
+        sage: H
+        Matching covered staircase graph: graph on 8 vertices
+        sage: H == G
+        True
+        sage: H.get_matching()
+        [(0, 1, None), (2, 7, None), (3, 6, None), (4, 5, None)]
+
+        sage: G = Graph({0: [1, 2, 3, 4], 1: [2, 5],
+        ....:            2: [5], 3: [4, 5], 4: [5]})
+        sage: H = MatchingCoveredGraph(G)
+        sage: H
+        Matching covered graph on 6 vertices
+        sage: H == G
+        True
+        sage: H.get_matching()
+        [(0, 4, None), (1, 2, None), (3, 5, None)]
+
+        sage: # needs networkx
+        sage: import networkx
+        sage: G = Graph(networkx.complete_bipartite_graph(12, 12))
+        sage: H = MatchingCoveredGraph(G)
+        sage: H
+        Matching covered graph on 24 vertices
+        sage: H == G
+        True
+        sage: H.get_matching()
+        [(4, 23, None), (6, 21, None), (11, 16, None), (7, 20, None), (10, 17, None), (1, 14, None), (2, 13, None), (8, 19, None), (9, 18, None), (3, 12, None), (5, 22, None), (0, 15, None)]
```

Comment:
you should sort the edges, for instance using `sorted(H.get_mathcing()`. Indeed, the order of the edges may vary and so the doctest may fail.

Furhtermore, could you improve the output like
```
        [(4, 23, None), (6, 21, None), (11, 16, None), (7, 20, None),
	 (10, 17, None), (1, 14, None), (2, 13, None), (8, 19, None),
	 (9, 18, None), (3, 12, None), (5, 22, None), (0, 15, None)]
```
and do the same each time you output edges in doctests.